### DISCUSSION TITLE
The Minimally Viable Product FDA 2.0

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/HEC-FDA#23] [Ideas]
We've had a lot of discussion over what a minimally viable product is, and what is our goal. To avoid this discussion becoming a vocab disagreement, I propose we accept my definitions below 

From Wikipedia-
Minimally Viable Product (MVP): A minimum viable product is a version of a product with just enough features to be **usable by early customers** who can then provide feedback for future product development.

New Jargon-
Guidance Conforming Product: The product release which at a minimum, provides all the necessary functionality for it's users to conform to USACE Guidance without relying at all on the previous version of FDA.

I wanted to separate these ideas out. Notably, MVP does not require full acceptance by our users, nor fulfilment of all USACE guidelines. It does need to be complete enough that a user could conceivably use it for SOMETHING, and provide feedback. Previously Richard and I have been working toward defining the Guidance Conforming Product and using the term MVP incorrectly by this definition, so I wanted to create a term for what we have been discussing to differentiate the ideas clearly.  

With that, what is our minimally viable product?

**Comment by rnugent3** [comment]
Brennan, I think your imagination is on point. I’ll write a concise list of what I think is the MVGCP this afternoon.

**Reply by rnugent3** [reply]
I just revisited our FDA Version 2.0 Requirements page on Confluence. I think that the requirements we have laid out identify the minimally viable guidance conforming product. 

https://www.hec.usace.army.mil/confluence/display/FDA/FDA+2.0+Requirements

**Comment by HenryGeorgist** [comment]
So I would say Unregulated Flow Frequency -> Rating Curve with uncertainty ->User defined Aggregated Stage Damage Curve -> EAD is a minimally viable product, and could be tested via the model logic with no V or VM interaction. 
Next we would want to connect it to a VM and test.
Then build a simple UI to trigger the compute from the VM down to the Model.
